core/src/main/java/hudson/Launcher.java

```diff
@@ -800,6 +800,10 @@ public Proc launch(ProcStarter ps) throws IOException {
             for ( int idx = 0 ; idx < jobCmd.length; idx++ )
             	jobCmd[idx] = jobEnv.expand(ps.commands.get(idx));
 
+            if (Functions.isWindows()) {
+                jobCmd = new ArgumentListBuilder(jobCmd).toWindowsCommand().toCommandArray();
```

Comment:
`toWindowsCommand()` delegates to `cmd.exe` so it should not be a problem IMHO, though I know little about Windows or cygwin and I have never heard about the POSIX subsystem for Win. Actually, the issue I reported appears when Cygwin is installed (I do not have machine without cygwin at hand). I have tested the fix successfully on oracle JDK 7 and 8 (with cygwin).